### PR TITLE
Fix: apply button is not replaced with check mark icon

### PR DESCRIPTION
**Pull request by FadhlanR**
**Problem**
I encountered a bug where there are multiple responses from ai bot and the patch command message is not the first response. The apply button of that message would not be replaced with check mark icon even after the patch was applied. 

**Solution**
We looked up the reaction event (the event that indicates the command is already applied) using `eventId` from the command payload while `eventId` in command payload can be `undefined`.  If `eventId` in command payload is undefined, then that message is not a replacement message for "Thinking" message, in that case we can use `eventId` of the message it self.

### CODE REVIEW

**Comment by tintinthong** [comment]
> I encountered a bug where there are multiple responses from ai bot and the patch command message is not the first response.

It looks like the order of this is fixed in this PR. Thx!

> The apply button of that message would not be replaced with check mark icon even after the patch was applied.

Hmmm does this PR fix the infinite circle spinner? I have actually found a situation where it doesn't. Particularly, when the module runs into an error after applying a json patch

**Comment by tintinthong** [comment]
> Hmmm does this PR fix the infinite circle spinner? I have actually found a situation where it doesn't. Particularly, when the module runs into an error after applying a json patch

Perhaps, this might be in a separate PR. But essentially applyButtonState is not tracking when failedCommandState is set EDIT: oh could it because of different event ids
EDIT: I think its correct its the same event id

File: packages/host/tests/integration/components/operator-mode-test.gts [hunk 1]
```diff
@@ -1499,6 +1499,63 @@ module('Integration | operator-mode', function (hooks) {
         .exists();
     });
 
+    test('essures applied state displayed as a check mark even eventId in command payload is undefined', async function (assert) {
```
Comment (jurgenwerk) [inline packages/host/tests/integration/components/operator-mode-test.gts]:
```suggestion
    test('assures applied state displayed as a check mark even eventId in command payload is undefined', async function (assert) {
```

**Review by jurgenwerk — APPROVED** [review]

**Comment by FadhlanR** [comment]
> > I encountered a bug where there are multiple responses from ai bot and the patch command message is not the first response.
> 
> It looks like the order of this is fixed in this PR. Thx!
> 
> > The apply button of that message would not be replaced with check mark icon even after the patch was applied.
> 
> Hmmm does this PR fix the infinite circle spinner? I have actually found a situation where it doesn't. Particularly, when the module runs into an error after applying a json patch

I had another PR to fix the order of those messages and I merged it today.

I think the infinite circle spinner is another issue because I didn't encounter it.

**Comment by tintinthong** [comment]
> > > I encountered a bug where there are multiple responses from ai bot and the patch command message is not the first response.
> > 
> > 
> > It looks like the order of this is fixed in this PR. Thx!
> > > The apply button of that message would not be replaced with check mark icon even after the patch was applied.
> > 
> > 
> > Hmmm does this PR fix the infinite circle spinner? I have actually found a situation where it doesn't. Particularly, when the module runs into an error after applying a json patch
> 
> I had another PR to fix the order of those messages and I merged it today.
> 
> I think the infinite circle spinner is another issue because I didn't encounter it.

Ahh ok. Ill file a new ticket then

**Review by tintinthong — APPROVED** [review]

**Pull request merged**